frame/elections-phragmen/src/lib.rs

```diff
@@ -115,6 +115,47 @@ type BalanceOf<T> =
 type NegativeImbalanceOf<T> =
 	<<T as Trait>::Currency as Currency<<T as frame_system::Trait>::AccountId>>::NegativeImbalance;
 
+
+/// Helper functions for migrations of this module.
+pub mod migrations {
+	use super::*;
+	use frame_support::migration::StorageKeyIterator;
+	use frame_support::Twox64Concat;
+
+	/// Migrate from the old legacy voting bond (fixed) to the new one (per-vote dynamic).
+	///
+	/// Will only be triggered if storage version is V1.
+	pub fn migrate_to_recorded_deposit<T: Trait>(old_deposit: BalanceOf<T>) -> Weight {
+		if <Module<T>>::pallet_storage_version() == StorageVersion::V1 {
+			let mut count = 0;
+			<StorageKeyIterator<T::AccountId, (BalanceOf<T>, Vec<T::AccountId>), Twox64Concat>>::new(
+				<Voting<T>>::module_prefix(),
+				b"Voting",
+			)
+			.for_each(|(who, (stake, votes))| {
+				// Insert a new value into the same location, thus no need to do `.drain()`.
+				let deposit = old_deposit;
+				let voter = Voter { votes, stake, deposit };
+				<Voting<T>>::insert(who, voter);
+				count += 1;
+			});
+
+			PalletStorageVersion::put(StorageVersion::V2PerVoterDeposit);
+			frame_support::debug::info!(
+				"🏛 pallet-elections-phragmen: {} voters migrated to V2PerVoterDeposit.",
+				count,
+			);
+			Weight::max_value()
```

Comment:
would be good to have a migration test IMHO

Reply:
Looking into it, and I already did one here: https://github.com/paritytech/substrate/pull/7040#issuecomment-694190522